src/mscorlib/src/System/Collections/Generic/List.cs

```diff
@@ -1051,6 +1054,29 @@ public void Sort(IComparer<T> comparer)
             return true;
         }
 
+        private class PooledIEnumerator : IEnumerator<T>
+        {
+            private Enumerator _enumerator;
+
+            internal PooledIEnumerator Initalize(List<T> list)
+            {
+                _enumerator = new Enumerator(list);
+                return this;
+            }
+            public bool MoveNext() => _enumerator.MoveNext();
+            public T Current => _enumerator.Current;
+            object IEnumerator.Current => Current;
+            public void Dispose()
+            {
+                _enumerator = default(Enumerator);
```

Comment:
This break existing code that accidentally uses an enumerator after `Dispose` has been called. In the existing implementation both `MoveNext` and `Current` continue to work after the enumerator is disposed. In the new implementation `MoveNext` throws `NullReferenceException` and `Current` returns `default(T)`.